erts/preloaded/src/erlang.erl

```diff
@@ -944,8 +945,9 @@ fun_info_mfa(_Fun) ->
     erlang:nif_error(undefined).
 
 %% fun_to_list/1
--spec erlang:fun_to_list(Fun) -> string() when
-      Fun :: function().
+-spec erlang:fun_to_list(Fun) -> String when
+      Fun :: function(),
+      String :: string().
```

Comment:
I see no reason to introduce String :: string() (a function constraint) for the return value.

When we introduce these extra function constraints, in most cases for arguments, it's to make it easier for the pretty-printer to create readable specs. In this case it's not necessary.

One could argue that it's not necessary to introduce a function constraint for the argument Fun either, but that's another topic.

Reply:
I think the constraint was there even before my change so I have not added any extra constraint. I introduced a name for the return value to make it easier to refer to it in the documentation. This is not necessary to do but I think it makes the documentation better. Btw, I'm currently rewriting the description of `erlang:fun_to_list/1` based on suggestions from @bjorng so it will change soon.

Reply:
You can also use an annotation, like "-> String :: string()".

Reply:
Ok, thanks, I did not know that.

Reply:
Fixed in 0d1d237